deepchem/models/torch_models/infograph.py

```diff
@@ -162,10 +177,183 @@ def forward(self, data):
+    The model randomly samples pairs of graphs and substructures, and then maximizes their mutual information by minimizing their distance in a learned embedding space.
+
+    This can be used for downstream tasks such as graph classification and molecular property prediction.It is implemented as a ModularTorchModel in order to facilitate transfer learning.
+
+    References
+    ----------
+    1. Sun, F.-Y., Hoffmann, J., Verma, V. & Tang, J. InfoGraph: Unsupervised and Semi-supervised Graph-Level Representation Learning via Mutual Information Maximization. Preprint at http://arxiv.org/abs/1908.01000 (2020).
+
+    Parameters
+    ----------
+    num_features: int
+        Number of node features for each input
+    edge_features: int
+        Number of edge features for each input
+    embedding_dim: int
+        Dimension of the embedding
+    num_gc_layers: int
+        Number of graph convolutional layers
+    prior: bool
+        Whether to use a prior expectation in the loss term
+    gamma: float
+        Weight of the prior expectation in the loss term
+    measure: str
+        The divergence measure to use for the unsupervised loss. Options are 'GAN', 'JSD',
+        'KL', 'RKL', 'X2', 'DV', 'H2', or 'W1'.
+    average_loss: bool
+        Whether to average the loss over the batch
+
+    Example
+    -------
+    >>> from deepchem.models.torch_models.infograph import InfoGraphModel
+    >>> from deepchem.feat import MolGraphConvFeaturizer
+    >>> from deepchem.data import NumpyDataset
+    >>> import torch
+    >>> smiles = ["C1CCC1", "C1=CC=CN=C1"]
+    >>> featurizer = MolGraphConvFeaturizer(use_edges=True)
+    >>> X = featurizer.featurize(smiles)
+    >>> y = torch.randint(0, 2, size=(2, 1)).float()
+    >>> w = torch.ones(size=(2, 1)).float()
+    >>> ds = NumpyDataset(X, y, w)
+    >>> num_feat = max([ds.X[i].num_node_features for i in range(len(ds))])
+    >>> edge_dim = max([ds.X[i].num_edge_features for i in range(len(ds))])
+    >>> model = InfoGraphModel(num_feat, edge_dim, 15)
+    >>> loss = model.fit(ds, nb_epoch=1)
+    """
+
+    def __init__(self,
+                 num_features,
+                 embedding_dim,
+                 num_gc_layers=5,
+                 prior=True,
+                 gamma=.1,
+                 measure='JSD',
+                 average_loss=True,
+                 **kwargs):
+        self.num_features = num_features
+        self.embedding_dim = embedding_dim * num_gc_layers
+        self.num_gc_layers = num_gc_layers
+        self.gamma = gamma
+        self.prior = prior
+        self.measure = measure
+        self.average_loss = average_loss
+        self.localloss = LocalMutualInformationLoss()._create_pytorch_loss(
+            measure, average_loss)
+        self.components = self.build_components()
+        self.model = self.build_model()
+        super().__init__(self.model, self.components, **kwargs)
+
+    def build_components(self) -> dict:
+        return {
```

Comment:
Can you add a table documenting the components and their names like you did for InfoGraphStar?